products/bash/entitled-registry.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+#******************************************************************************
+# Licensed Materials - Property of IBM
+# (c) Copyright IBM Corporation 2019. All Rights Reserved.
+#
+# Note to U.S. Government Users Restricted Rights:
+# Use, duplication or disclosure restricted by GSA ADP Schedule
+# Contract with IBM Corp.
+#******************************************************************************
+
+# PREREQUISITES:
+#   - Logged into cluster on the OC CLI (https://docs.openshift.com/container-platform/4.4/cli_reference/openshift_cli/getting-started-cli.html)
+#
+# PARAMETERS:
+#   -n : <namespace> (string), Defaults to 'cp4i'
+#
+#   With defaults values
+#     ./entitled-registry.sh
+#
+#   With overridden values
+#     ./entitled-registry.sh -n <namespace>
+
+function usage {
+  echo "Usage: $0 -n <namespace>"
+  exit 1
+}
+
+namespace="cp4i"
+
+while getopts "n:" opt; do
+  case ${opt} in
+    n ) namespace="$OPTARG"
+      ;;
+    \? ) usage; exit
+      ;;
+  esac
+done
+
+oc project ${namespace}
+echo "INFO: Namespace passed='${namespace}'"
+
+echo -e "\n----------------------------------------------------------------------------------------------------------------------------------------------------------\n"
+
+echo "INFO: Creating secret to pull base images from Entitled Registry"
+DOCKERCONFIGJSON_ER=$(oc get secret -n ${namespace} ibm-entitlement-key -o json | jq -r '.data.".dockerconfigjson"' | base64 --decode)
+if [ -z ${DOCKERCONFIGJSON_ER} ] ; then
+  echo "ERROR: Failed to find ibm-entitlement-key secret in the namespace '${namespace}'" 1>&2
+  exit 1
+fi
+
+export ER_REGISTRY=$(echo "$DOCKERCONFIGJSON_ER" | jq -r '.auths' | jq 'keys[]' | tr -d '"')
+export ER_USERNAME=$(echo "$DOCKERCONFIGJSON_ER" | jq -r '.auths."cp.icr.io".username')
+export ER_PASSWORD=$(echo "$DOCKERCONFIGJSON_ER" | jq -r '.auths."cp.icr.io".password')
+
+# Creating a new secret as the type of entitlement key is 'kubernetes.io/DOCKERCONFIGJSON' but we need secret of type 'kubernetes.io/basic-auth'
+# to pull imags from the ER
+cat << EOF | oc apply --namespace ${namespace} -f -
```

Comment:
I think this secret is only used for a pipeline service account? Maybe this could be moved to `configure-ocp-pipeline.sh`?